src/middlewared/middlewared/worker.py

```diff
@@ -46,6 +47,12 @@ def __init__(self):
         self.client = None
         self.logger = logging.getLogger('worker')
 
+    async def run_in_thread(self, method, *args, **kwargs):
+        with concurrent.futures.ThreadPoolExecutor() as executor:
```

Comment:
Using `with` here is dangerous. Imagine a blocking IO operation occurring in thread and then coroutine is cancelled (e.g. on timeout). This will block because ThreadPoolExecutor's `__exit__` will wait for thread shutdown. I've solved this issue like this: https://github.com/freenas/freenas/blob/master/src/middlewared/middlewared/main.py#L858